broadcast/src/buffered/engine.rs

```diff
@@ -171,7 +177,7 @@ impl<
                     };
 
                     // Decode the message
-                    let message = match M::decode(msg) {
+                    let message = match M::decode_cfg(msg, self.decode_config.clone()) {
```

Comment:
I recommend making this `decode_cfg(msg, &cfg)` (so we don't need to clone this object for each decode.